src/Symfony/Component/Console/Question/Question.php

```diff
@@ -85,6 +86,27 @@ public function setMultiline(bool $multiline): static
         return $this;
     }
 
+    /**
+     * Returns the timeout in seconds.
+     */
+    public function getTimeout(): ?int
+    {
+        return $this->timeout;
+    }
+
+    /**
+     * The timeout is the maximum time the user has to answer the question.
+     * If the user does not answer within this time, an exception will be thrown.
+     *
+     * @return $this
+     */
+    public function setTimeout(?int $seconds): static
```

Comment:
As `stream_select()` supports microseconds, what about supporting `float` here instead of `int`?

Reply:
As this targets user interaction, I think microseconds are unnecessary.